### ISSUE TITLE
[tx] Extract t1 from UFO fails but extracting cff works

### ISSUE THREAD

**Issue by NightFurySL2001** [adobe-type-tools/afdko#1632]
When passing the following UFO fonts into `tx -t1`, the program reports an error as below:
```
tx.exe: --- .\orig-pearl-tangyuan.ufo
tx.exe: (ufr) parsing failed
```
One is directly exported from FontLab, another (-2) is slightly modified. Both UFO are able to work with `tx -cff` and then `tx -t1`, but the glyph ordering is lost in the process.

Both fonts are able to be built with Google's `fontmake` so there shouldn't be any problem with the UFOs.

[orig-pearl-tangyuan.zip](https://github.com/adobe-type-tools/afdko/files/11112054/orig-pearl-tangyuan.zip)
[orig-pearl-tangyuan-2.zip](https://github.com/adobe-type-tools/afdko/files/11112055/orig-pearl-tangyuan-2.zip)

**Comment by punchcutter** [comment]
There's this contour way on the right side of uni8D9F that's causing it to fail. When converting to t1 it doesn't know what to do with that. 
<img width="1192" alt="uni8D9F" src="https://user-images.githubusercontent.com/11257204/229291264-33129207-7b23-4a01-951a-dd1d0dec3dd9.png">

Delete it and everything is fine. We should give a better error message since "parsing failed" doesn't really tell us anything .

**Comment by punchcutter** [comment]
If you converted with `-cff` first you should have seen the warning `tx: (cfw) no initial moveto (inserted) <uni8D9F>`

**Comment by frankrolf** [comment]
Ran into this today with a UFO that contained a single quadratic segment (among 3725 cubic segments).

The error message was `Command '['tx', '-t1', 'font.ufo', '/var/folders/7z/vl3dc3kd07bfz1q_5ybyk5wm0000gn/T/tmphui0y0jo']' died with <Signals.SIGSEGV: 11>.`, which is cryptic.

**Comment by kaydeearts** [comment]
Thanks for this submission! The issue Frank commented above was similar but exercised a different part of the code (and a fix for it was just merged into develop) but I've started investigating this specific issue as well

**Comment by punchcutter** [comment]
The original issue was a contour with a single point in it. Here is a much smaller test case: [single-point-contour-failure.ufo.zip](https://github.com/adobe-type-tools/afdko/files/11417782/single-point-contour-failure.ufo.zip)

The next issue was the mistake of converting it all to quadratic. Here is a much smaller test case: [has-quadratic-curves.ufo.zip](https://github.com/adobe-type-tools/afdko/files/11417781/has-quadratic-curves.ufo.zip)

**Comment by skef** [comment]
Waiting on PR
